test/xbgpu/test_engine.py

```diff
@@ -853,12 +855,15 @@ async def _send_data(
             for i, (corrprod_output, capture_stop_heap_index) in enumerate(
                 zip(corrprod_outputs, corrprod_capture_stop_heap_indices)
             ):
-                if capture_stop_heap_index > 0:
+                if capture_stop_heap_index is not None and capture_stop_heap_index > 0:
                     # Find out which accumulation the heap belongs to
-                    acc_index = capture_stop_heap_index // corrprod_output.heap_accumulation_threshold
+                    stop_acc_index = capture_stop_heap_index // corrprod_output.heap_accumulation_threshold
                     # Again, if data goes missing in an accumulation,
-                    # the accumulation is not transmitted.
-                    acc_counts[i][acc_index] = 0
+                    # this accumulation and all subsequent accumulations
+                    # are not transmitted.
+                    acc_indices_affected = filter(lambda acc_index: acc_index >= stop_acc_index, acc_counts[i])
```

Comment:
List comprehensions are preferred Python syntax over `filter` and `map`.

Reply:
Argh, I had originally typed up a list comprehension. I think I was trying to do too much in it, which is why I opted for this.
* Will update accordingly.